src/main/java/org/jenkinsci/plugins/workflow/graphanalysis/FlowScanningUtils.java

```diff
@@ -59,6 +62,42 @@ public boolean apply(FlowNode input) {
     // Default predicates, which may be used for common conditions
     public static final Predicate<FlowNode> MATCH_BLOCK_START = (Predicate)Predicates.instanceOf(BlockStartNode.class);
 
+    /** Sorts flownodes putting the one begun last (oldest startTime) at the end, with null times last
+     *  because likely they represent the newest nodes with a {@link TimingAction} not attached yet. */
+    public static final Comparator<FlowNode> TIME_ORDER_COMPARATOR = new Comparator<FlowNode>() {
+        /** Implements null checking because the use of this method will not easily permit FindBugs verification on NonNull*/
+        public int compare(@CheckForNull FlowNode first, @CheckForNull FlowNode second) {
+            if (first == null || second == null) {
+                return 0;  // Sorting by start time is 100% irrelevant
+            }
+            TimingAction timingFirst = first.getPersistentAction(TimingAction.class);
+            TimingAction timingSecond = second.getPersistentAction(TimingAction.class);
+            if (timingFirst != null && timingSecond != null) {
+                return Long.compare(timingFirst.getStartTime(), timingSecond.getStartTime());
+            } else if (timingFirst == null && timingSecond == null) {
+                return 0;
+            } else { // Only one is null, that one should return the greater value
+                return (timingSecond == null) ? -1 : 1 ;
+            }
+        };
+    };
+
+    public static final Comparator<FlowNode> ID_ORDER_COMPARATOR = new Comparator<FlowNode>() {
```

Comment:
There's a valid use case for iota-based sorting, even though Bismuth attempts to replace the inappropriate uses of this with the Time Order comparator or proper graph-driven logic. 

Implementing this means I can point people at it rather than watching them reinvent this particular wheel for the n-th time.